privacyidea/lib/resolvers/HTTPResolver.py

```diff
@@ -0,0 +1,247 @@
+        # Validations
+        getParam(config, 'method', optional=False, allow_empty=False, allowed_values=('get', 'post', 'GET', 'POST'))
+        getParam(config, 'endpoint', optional=False, allow_empty=False)
+        getParam(config, 'requestMapping', optional=False, allow_empty=False)
+        getParam(config, 'responseMapping', optional=False, allow_empty=False)
+        hasSpecialErrorHandler = getParam(
+            config,
+            'hasSpecialErrorHandler',
+            optional=False,
+            default=False,
+            allow_empty=False
+        )
+
+        # Validate special error handler
+        if hasSpecialErrorHandler and not config.get('errorResponse'):
+            getParam(config, 'errorResponse', optional=False, allow_empty=False)
+
+        self.config = config
+        return self
+
+    @classmethod
+    def testconnection(cls, param):
+        """
+        This function lets you test if the parameters can be used to create a
+        working resolver. Also, you can use it anytime you see if the API is
+        running as expected.
+        The implementation should try to make a request to the HTTP API and verify
+        if user can be retrieved.
+        In case of success it should return the raw http response.
+
+        :param param: The parameters that should be saved as the resolver
+        :type param: dict
+        :return: returns True in case of success and a raw response
+        :rtype: tuple
+        """
+        desc = ""
+        success = False
+        try:
+            resolver = HTTPResolver()
+            resolver.loadConfig(param)
+            response = resolver._getUser(param.get('testUser'))
+            desc = response
+            success = True
+        except Exception as e:
+            success = False
+            desc = "failed: {0!s}".format(e)
+        return success, desc
+
+    #
+    #   Private methods
+    #
+    def _getUser(self, userid):
+        param = self.config
+        method = param.get('method').lower()
+        endpoint = param.get('endpoint')
+        requestMappingJSON = json.loads(param.get('requestMapping').replace("{userid}", userid))
+        responseMapping = json.loads(param.get('responseMapping'))
+        headers = json.loads(param.get('headers', '{}'))
+        hasSpecialErrorHandler = bool(param.get('hasSpecialErrorHandler'))
+        errorResponse = json.loads(param.get('errorResponse'))
+
+        if method == "post":
+            httpResponse = requests.post(endpoint, json=requestMappingJSON, headers=headers)
+        else:
+            httpResponse = requests.get(endpoint, urlencode(requestMappingJSON), headers=headers)
+
+        # Raises HTTPError, if one occurred.
+        httpResponse.raise_for_status()
+
+        jsonHTTPResponse = httpResponse.json()
+
+        if hasSpecialErrorHandler:
+            # verify if error response mapping is a subset of the json http response
+            if errorResponse.items() <= jsonHTTPResponse.items():
```

Comment:
i think we should use a `set` here for comparison:
`if errorResponse and set(errorResponse.items()) <= set(jsonHTTPResponse.items()):`

Reply:
Ok, so this doesn't work if there are embedded dictionaries. I'll look for something else.

Reply:
the following works:
`all([x in jsonHTTPResponse.items() for x in errorResponse.items()])`
but we still need to check if `errorResponse` is empty.

Reply:
I've changed with your implementation and looks great!

> but we still need to check if errorResponse is empty.

Here's https://github.com/privacyidea/privacyidea/pull/2083/files#diff-7d7a63c6aac770da073091e1bb88a464R177 :)